kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/ErrorTransform.java

```diff
@@ -0,0 +1,383 @@
+  private static final String KEY_FAILURE = "KEY_CONVERTER";
+  private static final String VALUE_FAILURE = "VALUE_CONVERTER";
+  private static final String HEADER_FAILURE = "HEADER_CONVERTER";
+  private static final String SMT_FAILURE = "SMT_FAILURE";
+
+  private TransformExceptionHandler converterErrorHandler;
+  private TransformExceptionHandler smtErrorHandler;
+  private List<Transformation<SinkRecord>> smts;
+  private Function<SinkRecord, SchemaAndValue> keyConverter;
+  private Function<SinkRecord, SchemaAndValue> valueConverter;
+  private Function<SinkRecord, Headers> headerConverterFn;
+
+  public static final ConfigDef CONFIG_DEF =
+      new ConfigDef()
+          .define(
+              KEY_CONVERTER,
+              ConfigDef.Type.STRING,
+              "org.apache.kafka.connect.converters.ByteArrayConverter",
+              ConfigDef.Importance.MEDIUM,
+              "key.converter")
+          .define(
+              VALUE_CONVERTER,
+              ConfigDef.Type.STRING,
+              null,
+              ConfigDef.Importance.MEDIUM,
+              "value.converter")
+          .define(
+              HEADER_CONVERTER,
+              ConfigDef.Type.STRING,
+              "org.apache.kafka.connect.converters.ByteArrayConverter",
+              ConfigDef.Importance.MEDIUM,
+              "header.converter")
+          .define(TRANSFORMATIONS, ConfigDef.Type.STRING, null, ConfigDef.Importance.MEDIUM, "smts")
+          .define(
+              CONVERTER_ERROR_HANDLER,
+              ConfigDef.Type.STRING,
+              "io.tabular.iceberg.connect.transforms.DefaultExceptionHandler",
+              ConfigDef.Importance.MEDIUM,
+              "Error handling class for converter errors")
+          .define(
+              SMT_ERROR_HANDLER,
+              ConfigDef.Type.STRING,
+              "io.tabular.iceberg.connect.transforms.DefaultExceptionHandler",
+              ConfigDef.Importance.MEDIUM,
+              "Error handling class for SMT errors");
+
+  @Override
+  public SinkRecord apply(SinkRecord record) {
+    // tombstones returned as-is
+    if (record == null || record.value() == null) {
+      return record;
+    }
+
+    DeserializedRecord deserialized = deserialize(record);
+    if (deserialized.isFailed()) {
+      return deserialized.getRecord();
+    }
+
+    SinkRecord transformedRecord = deserialized.getRecord();
+
+    for (Transformation<SinkRecord> smt : smts) {
+      try {
+        transformedRecord = smt.apply(transformedRecord);
+        if (transformedRecord == null) {
+          break;
+        }
+      } catch (Exception e) {
+        return smtErrorHandler.handle(record, e, SMT_FAILURE);
+      }
+    }
+    // SMT could filter out messages
+    if (transformedRecord == null) {
+      return null;
+    }
+
+    return newRecord(record, transformedRecord);
+  }
+
+  @Override
+  public ConfigDef config() {
+    return CONFIG_DEF;
+  }
+
+  @Override
+  public void close() {}
```

Comment:
My bad.  Forgot they implement autocloseable.  Thanks

Reply:
I have to otherwise it's a compiler error. 

If I turn it off 

```
ErrorTransform is not abstract and does not override abstract method close() in Transformation
public class ErrorTransform implements Transformation<SinkRecord> {
```